router/core/graphql_handler.go

```diff
@@ -375,9 +375,16 @@ func addErrorToSpan(ctx context.Context, err error) {
 	rtrace.AttachErrToSpan(span, err)
 }
 
-func propagateSubgraphErrors(ctx *resolve.Context) {
+func propagateSubgraphErrors(ctx *resolve.Context, requestLogger *zap.Logger) {
+	// Get subgraph errors
 	err := ctx.SubgraphErrors()
+	// Add error to span
 	addErrorToSpan(ctx.Context(), err)
+	// Check if there are subgraph errors
+	if ctx.SubgraphErrors() != nil {
```

Comment:
Please reuse the result from the previous execution and you can put `addErrorToSpan(ctx.Context(), err)` inside the if branch, only when an err is set.

Reply:
I'm sorry ... I totally forgot to edit my code...